src/components/Game.js

```diff
@@ -15,108 +16,194 @@ class Game extends Component {
+    if (oldPitch !== pitch ){ // pitch has changed, A or B
+      this.setState({userPlayingPitch: pitch}, ()=>{devLogger(`pitch changed to: ${pitch}`)});
+      this.handlePitchChange(pitch);
+    } else { // pitch has not changed, C
+      this.handlePitchNoChange(pitch);
+    };
+  }
+
+  checkPitchInRange = (pitch) => {
+    if (pitch <= this.state.targetFreqRangeUpper && pitch >= this.state.targetFreqRangeLower) {return true} else {return false};
+  };
+
+  checkTimerStarted = () => {
+    if (this.state.targetTime) {return true} else {return false};
+  }
 
-    if (oldPitch !== pitch ){
-      this.setState({userPlayingPitch: pitch}, ()=>{
-        // console.log('in game, userPlayingPitch: ', this.state.userPlayingPitch);
-        if (pitch <= this.state.targetFreqRangeUpper && pitch >= this.state.targetFreqRangeLower) {
-          this.toggleNoteMatched();
+  handlePitchChange = (pitch) => {
+    let pitchInRange = this.checkPitchInRange(pitch); // true or false
+    let timerStarted = this.checkTimerStarted();
+    let targetTimeReached = this.checkTargetTimeReached();
+
+    if (pitchInRange) { // A
+      if (timerStarted) { // A1
+        if (targetTimeReached) { // A1a
+          this.generateSparkles();
+        } else { // A1b
+          devLogger('keep playing');
         }
-      });
+      } else { // A2
+          this.setTargetTime();
+      }
+
+    } else { // B note is out of target range ("wrong") or null
+      if (timerStarted) { // B1
+        if (targetTimeReached) { // B1a note changed to null or wrong note, timer has finished
+          this.handleSuccessfulRound();
+        } else { // B1b
+          this.voidTargetTime();
+          devLogger('TIMER VOIDED (B1b)');
+        }
+      } else { // B2
+        // do nothing
+        return;
+      }
     }
 
-    // add an else here to reset state to false if the time requirement isn't reached
+  };
+
+  handlePitchNoChange(pitch) { // C
+    if (this.state.targetTime && this.checkTargetTimeReached()) { // C1
+      this.generateSparkles();
+      devLogger('sparkles in C1');
+    } else { // C2
+      if (pitch) {
+        devLogger(`Keep going, you're SO CONSISTENT!`);
+      }
+    }
+  };
+
+
+  // GAME LOGIC HELPERS
+  generateSparkles = () => {
+    devLogger('* ... sparkles ... * success');
+  };
+
+  handleSuccessfulRound = () => {
+    this.props.increaseNotesPlayedCorrectlyCallback();
+    this.setNewNote();
+    this.voidTargetTime();
+    devLogger(`You got a point! Here's a new note.`);
   }
 
-  toggleNoteMatched = () => {
-    // this.state.noteMatched ? this.setState({noteMatched: false}) : this.setState({noteMatched: true})
-    this.setState({noteMatched: true},()=>{
-      console.log(this.state.noteMatched);
-    });
+  checkTargetTimeReached = () => {
+    const currentTime = new Date().getTime();
+    const targetTime = this.state.targetTime;
+
+    if (currentTime >= targetTime) {return true} else {return false};
```

Comment:
`return currentTime >= targetTime`

Reply:
Ahhh thank you! I knew there was a way to simplify this and couldn't remember in the moment.